llarp/net/route.cpp

```diff
@@ -338,28 +340,28 @@ namespace llarp::net
   {
     std::vector<std::string> gateways;
 #ifdef __linux__
-
-    FILE* p = popen("ip route", "r");
-    if (p == nullptr)
-      return gateways;
-    char* line = nullptr;
-    size_t len = 0;
-    ssize_t read = 0;
-    while ((read = getline(&line, &len, p)) != -1)
-    {
-      std::string line_str(line, len);
-      std::vector<std::string> words;
-      std::istringstream instr(line_str);
-      for (std::string word; std::getline(instr, word, ' ');)
+    // return true when it's all zeros
+    const auto is_zero = [](const auto& str) {
+      for (const auto& ch : str)
       {
-        words.emplace_back(std::move(word));
+        if (ch != '0')
+          return false;
       }
-      if (words[0] == "default" and words[1] == "via" and words[3] == "dev" and words[4] != ifname)
+      return true;
+    };
+    std::ifstream inf("/proc/net/route");
+    for (std::string line; std::getline(inf, line);)
+    {
+      const auto parts = split(std::string_view{line}, '\t');
```

Comment:
The explicit construction here isn't needed, `std::string`s are always implicitly convertible to `std::string_view`s (which is part of what makes string_view arguments nice).

Reply:
done